### DISCUSSION TITLE
Web3Modal With SignAPI Metamask only supported with QR Code

### DISCUSSION THREAD

**Discussion by 0xtchandragupta** [WalletConnect/walletconnect-monorepo#2646] [Developer Support]
I am using Web3Modal with Sign API to connect to an external wallet from my app.
https://docs.walletconnect.com/2.0/web/web3modal/react/sign-api/installation

The default modal that pops up does not have metamask or trustwallet, etc.
I tried to use explorerrecommendedwalletids in modalOptions as mentioned here https://docs.walletconnect.com/2.0/web/web3modal/react/sign-api/options#explorerrecommendedwalletids-optional 
to get metamask in the modal connect options. That did show up Metamask.
<img width="356" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/94593328/7d33521e-7618-4f14-abb5-9d0a15fdca07">

But on clicking it, it only shows the QR code option instead of triggering the meta mask extension which is already installed on my browser..
<img width="346" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/94593328/853a71a1-b5cc-43a0-a809-ec408e95d526">

I'm trying to add desktop wallets option in modalOptions as mentioned here
https://docs.walletconnect.com/2.0/web/web3modal/react/sign-api/options#desktopwallets-optional
to try to make it invoke the browser extension instead of showing QR code, but unsure what the values for native and universal links be.

Please help.

**Comment by glitch-txs** [comment]
You should use any of the options that use wagmi for this.

**Reply by 0xtchandragupta** [reply]
@glitch-txs 
But the wagmi one is not supporting message Signing.
Sign API seems to be the one to use for message Signing.
I want to get the user to sign a message from their wallet and it should work on any platform(desktop browser, mobile browser, mobile app).

Trying to do this in a react native app so.

**Reply by glitch-txs** [reply]
ahh for react native should use this one
https://docs.walletconnect.com/2.0/reactnative/web3modal/about

**Reply by 0xtchandragupta** [reply]
@glitch-txs 
With react-native the problem is that the list in the modal does not have Metamask or TrustWallet, which is the most important.
Check this https://github.com/orgs/WalletConnect/discussions/2650

**Comment by liubin595338764** [comment]
@glitch-txs  I have also encountered this problem. May I ask how should I configure Metamask inside? Just like the picture below, the documentation does not teach me how to configure the parameters. Where can I obtain such parameters? Can you provide an example of configuring Metamask? This is really needed.

https://docs.walletconnect.com/2.0/web/walletConnectModal/modal/options#desktopwallets-optional

<img width="351" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/5274428/062cf982-d991-4116-8433-d420458e95ee">

**Reply by glitch-txs** [reply]
The only way is using one of the packages with wagmi of Web3modal
https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/installation

**Reply by liubin595338764** [reply]
@glitch-txs I have also tried using Web3modal, but I couldn't find a provider for new Web3. We are using web3.js + Vue instead of React. Can you help me with how to use Web3modal+web3.js+HTML ?

**Reply by glitch-txs** [reply]
There's no such solution at the moment since wagmi uses viem instead of web3.js

You can handle MetaMask separately or migrate to wagmi

I've written a library that solves this but it's new and not official, you could also build your own solution.

WalletConnect team is working on a solution for this but not available atm.
